<a href="https://colab.research.google.com/github/RandDahmash/Mining-project/blob/main/phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project Goal:
The project's main goal is a classification and clustring task. We will build a machine learning model to predict a student's adaptability level in an online education environment. This will help identify the factors that influence whether a student will have a low, moderate, or high level of adaptability. The ultimate aim is to use these insights to help educational institutions and policymakers improve online learning resources and support systems.



# Data Source Link :
https://www.kaggle.com/datasets/mdmahmudulhasansuzan/students-adaptability-level-in-online-education

Data is from Kaggle


## Dataset Description
- Number and data types of attributes (features)  
- Number of records (instances)  
- Class attribute with its values and counts  
- Sample of the dataset

In [ ]:
import pandas as pd
data=pd.read_csv('Raw_dataset.csv')



target="Adaptivity Level"
all_columns = list(data.columns)
feature_columns = [c for c in all_columns if c !=target]


print("Feature columns in the database:")
print(feature_columns)
print("\nData types of features:")
print(data[feature_columns].dtypes)



print("Number of columns [excluding class]=",len(feature_columns))
print("The class attribute is : ",target)
print("Number of columns [including class]=",len(all_columns))




numOfRecords = len(data)
print("Number of records:", numOfRecords)



print("Class attribute is ",target)
print("Counts/label:")
print(data[target].value_counts().to_frame())


print("Sample of the database")
print(data.head(5))



Feature columns in the database:
['Gender', 'Age', 'Education Level', 'Institution Type', 'IT Student', 'Location', 'Load-shedding', 'Financial Condition', 'Internet Type', 'Network Type', 'Class Duration', 'Self Lms', 'Device']

Data types of features:
Gender                 object
Age                    object
Education Level        object
Institution Type       object
IT Student             object
Location               object
Load-shedding          object
Financial Condition    object
Internet Type          object
Network Type           object
Class Duration         object
Self Lms               object
Device                 object
dtype: object
Number of columns [excluding class]= 13
The class attribute is :  Adaptivity Level
Number of columns [including class]= 14
Number of records: 1205
Class attribute is  Adaptivity Level
Counts/label:
                  count
Adaptivity Level       
Moderate            625
Low                 480
High                100
Sample of the database
 

## Summary
The dataset contains 1,205 records and 14 attributes.  
The target variable is Adaptivity Level, which has three classes: Low, Moderate, High.  
This dataset description provides the foundation for building machine learning models in the next phases.